src/System.IO.FileSystem/src/System/IO/Directory.cs

```diff
@@ -63,24 +63,18 @@ internal static string EnsureTrailingDirectorySeparator(string fullPath)
         //
         public static bool Exists(string path)
         {
-            try
+            if (string.IsNullOrEmpty(path) || path.Contains('\0'))
```

Comment:
Why are we checking for `'\0'` here?
a) Isn't that a valid (though strange) character to use on Unix?
b) This is such a rare case, it doesn't seem like we should be doing a fast-path check for it.

Reply:
In the previous version also, we were throwing the argument exception if the path contains an embedded null

Reply:
https://github.com/dotnet/coreclr/blob/master/src/mscorlib/shared/System/IO/Path.Unix.cs#L26

Reply:
> we were throwing the argument exception if the path contains an embedded null

Ok